### PR TITLE
removed Cart route, controller, model and specs: dead code

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Related to #2523 

This beautiful part of OFN is 5 years old, this was used in the store frontend in the past.
I suspect it's just dead code.
(we may soon get back some of this to re-implement Order and OrderPopulator outside of Spree).

#### What should we test?
Shopping process, add to cart and checkout. Nothing should change.

#### Release notes
Removed Cart related old dead code.

Changelog Category: Removed

#### How is this related to the Spree upgrade?
This is related to the refactoring of the OrderPopulator out of Spree. #2523 is part of the Spree Upgrade process.

### CODE REVIEW

**Comment by luisramos0** [comment]
313 lines of code deleted, the build is green. Looking good. I wonder what do reviewers think? :-D

My only concern is: do we have external systems/users of these /cart endpoints? I don't see any other possible problem. I may be wrong.

File: app/models/cart.rb [hunk 1]
```diff
@@ -1,28 +0,0 @@
-class Cart < ActiveRecord::Base
```
Comment (mkllnk) [inline app/models/cart.rb]: I think removing the model should also go with a migration to remove the table. It also needs to drop the relationship between order and cart.

I just checked our production server for carts and our table is empty. So this should be safe.

I found several uses of `order.cart?`, but I suspect that it's testing whether the order is in the `cart` state and not if it has a cart. Can you double check that?

Comment (luisramos0) [reply app/models/cart.rb]: I doubled checked, yes, cart is one of the states of the order state machine.
All I have found on my double checking was more dead code, please see the new commit.

**Comment by Matt-Yorkley** [comment]
Staged on https://staging.openfoodnetwork.org.uk

**Comment by daniellemoorhead** [comment]
@myriamboure @RachL is one of you able to test this on UK staging when you're back please? 😃

**Comment by RachL** [comment]
Everything seems fine.

Here are my testing notes:
https://docs.google.com/document/d/1WXC89l-h8DUDbNBUUayb4Ox2JhrakatkeASU_lEC7_M/edit#

**Comment by luisramos0** [comment]
oh man, I love this one! first thing I change on the OFN database is a drop table!!!

**Pull request merged**